src/toctype.c

```diff
@@ -103,65 +107,29 @@ class ToCtypeVisitor : public Visitor
 
     void visit(TypeDelegate *t)
     {
-        t->ctype = type_delegate(Type_toCtype(t->next));
+        ctype = type_delegate(Type_toCtype(t->next));
     }
 
     void visit(TypeStruct *t)
     {
         //printf("TypeStruct::toCtype() '%s'\n", t->sym->toChars());
-        Type *tm = t->mutableOf();
-        if (tm->ctype)
```

Comment:
Just a minor refactoring.
In the old version: If the unqualified version of this type already has a ctype, we copy it and add the qualifiers.  Otherwise we create a new Classsym/ctype.

In the new version, we generate a new ctype if this is the unqualified version, otherwise recurse to get the unqualified ctype then copy it.

As far as I can tell, the old code would generate a symbol with the wrong qualifiers when called with a qualified before being called on the unqualified symbol, because both would be generated and set to the same value but the 'add modifiers' code would never be reached.